platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -938,12 +949,36 @@ def _parse_vpd_data(self, filename):
                 return result
 
             result = utils.read_key_value_file(filename, delimeter=": ")
-                
+
         except Exception as e:
             logger.log_error("Fail to decode vpd_data {} due to {}".format(filename, repr(e)))
 
         return result
 
+    def _get_spectrum_version(self):
+        """
+        Returns spectrum version of the platform
+
+        Returns:
+            Returns spectrum version of the platform
+        """
+        out = check_output(SPC_RETRIEVAL, shell=True)
```

Comment:
Check how you can reduce the mock to only mock this function, so you could get better UT covrage